beater/beater_test.go

```diff
@@ -0,0 +1,152 @@
+package beater
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	// make sure processors are loaded
+	_ "github.com/elastic/apm-server/include"
+
+	"github.com/elastic/apm-server/tests"
+	"github.com/elastic/beats/libbeat/monitoring"
+	"github.com/elastic/beats/libbeat/outputs"
+	pubs "github.com/elastic/beats/libbeat/publisher"
+	"github.com/elastic/beats/libbeat/publisher/pipeline"
+	"github.com/elastic/beats/libbeat/publisher/queue"
+	"github.com/elastic/beats/libbeat/publisher/queue/memqueue"
+)
+
+/*
+Run the benchmarks as follows:
+
+    $ go test beater/*.go -run=XXX -bench=. -cpuprofile=cpu.out
+
+then load the cpu profile file:
+
+		 $ go tool pprof beater.test cpu.out
```

Comment:
Indentation is off (also below). Probably tabs vs spaces something. Not sure if this would have an affect on go docs.

Reply:
thanks, fixed now. Not sure how that happened 🙄